traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices_required_capabilities.go

```diff
@@ -259,6 +268,62 @@ func (rc *RequiredCapability) Create() (error, error, int) {
 	return nil, nil, http.StatusOK
 }
 
+func (rc *RequiredCapability) ensureDSServerCap() (error, error, int) {
+	tx := rc.APIInfo().Tx
+
+	// Get assigned DS server IDs
+	dsServerIDs := []int64{}
+	if err := tx.Tx.QueryRow(`
+	SELECT ARRAY(
+		SELECT server 
+		FROM deliveryservice_server 
+		WHERE deliveryservice=$1
+	)`, rc.DeliveryServiceID).Scan(pq.Array(&dsServerIDs)); err != nil && err != sql.ErrNoRows {
+		return nil, fmt.Errorf("reading delivery service %v servers: %v", *rc.DeliveryServiceID, err), http.StatusInternalServerError
+	}
+
+	if len(dsServerIDs) == 0 { // no attached servers can return success right away
+		return nil, nil, http.StatusOK
+	}
+
+	// Get servers IDs that have the new capability
+	capServerIDs := []int64{}
+	if err := tx.QueryRow(`
+	SELECT ARRAY(
+		SELECT server
+		FROM server_server_capability 
+		WHERE server IN (
+			SELECT server 
```

Comment:
Technically you already have the answer to this subquery via the query above and could pass that data into this query, but I wouldn't hold off merging this for that reason.